File: Tagx00.Backend/src/main/java/trapx00/tagx00/vo/mission/text/TextMissionProperties.java
package trapx00.tagx00.vo.mission.text;

import trapx00.tagx00.entity.mission.textmissionsettings.TextMissionSetting;
import trapx00.tagx00.publicdatas.mission.MissionType;
import trapx00.tagx00.vo.mission.missiontype.MissionProperties;

import java.util.List;

public class TextMissionProperties extends MissionProperties {

    private List<TextMissionSetting> settings;
    private List<TextMissionType> textMissionTypes;

    public List<TextMissionType> getTextMissionTypes() {
        return textMissionTypes;
    }

    public void setTextMissionTypes(List<TextMissionType> textMissionTypes) {
        this.textMissionTypes = textMissionTypes;
    }

    public TextMissionProperties() {
    }

    public TextMissionProperties(List<TextMissionSetting> settings) {

        this.settings = settings;
        for (TextMissionSetting a : settings) {
            textMissionTypes.add(a.getTextMissionType());
        }
    }

    public TextMissionProperties(MissionType type, List<TextMissionSetting> settings) {
        super(type);
        this.settings = settings;
    }

    public List<TextMissionSetting> getSettings() {
        return settings;
    }

    public void setSettings(List<TextMissionSetting> settings) {
        this.settings = settings;
    }
}

File: Tagx00.Frontend/src/app/models/mission/text/TextMissionProperties.ts
import { MissionProperties } from "../MissionProperties";

export enum TextMissionType {
  KEYWORDS = "KEYWORDS",
  CLASSIFICATION = "CLASSIFICATION"
}


export interface TextMissionSetting {
  textMissionType: TextMissionType;
}

export interface TextMissionClassificationSetting extends TextMissionSetting{
  textMissionType: TextMissionType.CLASSIFICATION;
  classes: string[];
}

export interface TextMissionKeywordsSetting extends TextMissionSetting {
  textMissionType: TextMissionType.KEYWORDS;
  keywords: string[];
}

export interface TextMissionProperties extends MissionProperties {
  settings: TextMissionSetting[];
}
